src/animation/friction.ts

```diff
@@ -66,79 +64,107 @@ function reversePlayState(
-  onComplete,
-  infinite,
-}: Friction1DParams): { controller: Controller } {
-  const state: FrictionState = {
-    mover: {
-      mass: config.mass,
-      acceleration: [0, 0],
-      velocity: [config.initialVelocity, 0],
-      position: [0, 0],
+export function frictionGroup(animatingElement: FrictionAnimatingElement) {
+  // Take the initial animating element configuration and extend it with some state properties.
+  const statefulAnimatingElement: StatefulFrictionAnimatingElement = {
+    ...animatingElement,
+    state: {
+      mover: {
+        mass: animatingElement.config.mass,
+        acceleration: [0, 0],
+        velocity: [animatingElement.config.initialVelocity, 0],
+        position: [0, 0],
+      },
+      playState: PlayState.Forward,
+      maxDistance: getMaxDistanceFriction({
+        mu: animatingElement.config.mu,
+        initialVelocity: animatingElement.config.initialVelocity,
+      }),
+      complete: false,
+      paused: !!animatingElement.pause,
+      delayed: !!animatingElement.delay,
     },
-    playState: PlayState.Forward,
   };
 
-  const maxDistance = getMaxDistanceFriction({
-    mu: config.mu,
-    initialVelocity: config.initialVelocity,
-  });
-
-  const listener: Listener = (timestamp, lastFrame, stop) => {
-    /**
-     * Determine the number of milliseconds elapsed between the current frame
-     * and the last frame. If more than four frames have been dropped, assuming
-     * a 60fps interval, just use the timestamp of the current frame.
-     */
-
-    // Obtain the timestamp of the last frame. If this is the first frame, use the current frame timestamp.
-    let lastTime = lastFrame !== undefined ? lastFrame : timestamp;
-
-    /**
-     * If more than four frames have been dropped since the last frame,
-     * just use the current frame timestamp.
-     */
-    if (timestamp > lastTime + 64) {
-      lastTime = timestamp;
+  // If the element is not in the Set...
+  if (!animatingElements.has(statefulAnimatingElement)) {
+    // If it has a delay, set a timer to clear delayed state.
+    if (statefulAnimatingElement.state.delayed) {
+      setTimeout(() => {
+        statefulAnimatingElement.state.delayed = false;
+      }, statefulAnimatingElement.delay);
     }
 
-    // Determine the number of steps between the current frame and last recorded frame.
-    const steps = Math.floor(timestamp - lastTime);
+    animatingElements.add(statefulAnimatingElement);
+  }
 
-    // Apply the force of friction once for each step.
-    for (let i = 0; i < steps; i++) {
-      if (infinite) {
-        reversePlayState(state, config, maxDistance);
+  let startFn: () => void = () => {};
+  let pauseFn: () => void = () => {};
+  let stopFn: (element: StatefulFrictionAnimatingElement) => void = () => {};
+
+  // Only start the frameloop if there are elements to animate.
+  if (animatingElements.size > 0) {
+    const { start, stop } = rAF();
+
+    startFn = () => {
+      if (!isFrameloopActive) {
```

Comment:
Here's where we check if there is an already executing `rAF` setup. If so, no need to `start` the frameloop again, as each call to `update` will "see" new elements added to our `Set` of animating elements and ensure they start moving. Verified in Performance tabs on Chrome and FF that we only get one `rAF` call for n elements!